### ISSUE TITLE
[Bug] CorsixTH Campaign Map Four Corners can't be cleared despite fulfilling winning conditions

### ISSUE THREAD

**Issue by xyressa** [CorsixTH/CorsixTH#2675]
### Describe the issue

I was addicted enough to try out the custom ChrizmanTV campaign that came with the Corsix download, and currently am stuck on the Four Corners level. The briefing instructions at the start of the game says I need (1) Cure 600 people (2) Treat 75% of them (3) Have $200K (4) Value worth $300,000 and (5) 750 Reputation to win. But in the attached 'Charts' screenshots, I have clearly fulfilled all the win conditions including one new one that says to have killed <25%, but the level still hasn't cleared for me.

Is there a bug with this?

(I definitely cleared the value one as I bought a new plot and built a gigantic psych room with like, 100 skeletons.)

![FourCorners_map](https://github.com/user-attachments/assets/d65bee57-1c55-444c-b459-ec99a67c55ec)
![FourCorners_charts](https://github.com/user-attachments/assets/97801985-ddd2-42f3-8604-63cc8bbc15a3)
![FourCorners_briefing](https://github.com/user-attachments/assets/81245102-e067-4390-9c4d-b0a574091dbd)

### Steps to Reproduce

Fulfilled the winning conditions for the level 'Four Corners' in the ChizmanTV custom campaign but the stage does not clear

### Expected Behaviour

Level should auto-clear as the winning conditions are met.

### Have you enabled any current CorsixTH Experimental Features?

- [ ] Right mouse panning
- [ ] Falling actions

### Save Game

Save game attached to report: Yes
[FourCorners.zip](https://github.com/user-attachments/files/17406975/FourCorners.zip)

### CorsixTH Version

v0.68.0 RC 2

### Operating System

Windows 10 Home

### Theme Hospital Version

GOG.com

### Gamelog.txt

```yaml
Gamelog generated on 2024-10-05 14:31:49
Compiled with api_version: 2687, arch: x64, audio: true, os: windows, update_check: true
SDL renderer: direct3d
Lua 5.4 run with api version: 2687, game version: v0.68.0 RC 2, savegame version: 193
Created game with savegame version 193.
Warning: Trying to resolve door deadlock at (65, 90)
Error in timer handler: 
Lua\calls_dispatcher.lua:436: call to be executed is still assigned
stack traceback:
	[C]: in function 'assert'
	Lua\calls_dispatcher.lua:436: in method 'executeCall'
	Lua\calls_dispatcher.lua:364: in method 'answerCall'
	Lua\humanoid_actions\meander.lua:42: in field '?'
	Lua\entities\humanoid.lua:476: in method 'startAction'
	Lua\entities\humanoid.lua:577: in method 'finishAction'
	Lua\humanoid_actions\walk.lua:187: in function <Lua\humanoid_actions\walk.lua:172>
	(...tail calls...)
	Lua\entity.lua:216: in field 'tick'
	Lua\entities\humanoids\staff.lua:112: in field 'tick'
	Lua\entities\humanoids\staff\doctor.lua:95: in method 'tick'
	Lua\world.lua:917: in method 'onTick'
	Lua\app.lua:1239: in function <Lua\app.lua:1236>
	(...tail calls...)
	Lua\app.lua:1139: in function <Lua\app.lua:1134>
```

**Comment by Matroftt** [comment]
It is the issue of losing condition has been implemented poorly for this level, to beat the level you need ~$250,000 hospital value, but due to "do not kill x% of patients" it is not showing

**Comment by Matroftt** [comment]
To fix it, lose condition should show only when 10% of patients is killed or more

**Comment by ARGAMX** [comment]
Agree with [Matroftt](https://github.com/Matroftt).
This hospital don't hit the value requirement.
This can be seen in the annual report.

![Screenshot 2024-10-17 at 11 04 32](https://github.com/user-attachments/assets/d5311fcb-db0d-4255-8ada-552853fdc1e4)

From my point of view Unfulfilled conditions should be prioritized over fulfilled ones for display.

**Comment by xyressa** [comment]
Ahh that makes sense, I was trying to figure out where I could see my hospital's value since I'm used to it showing up on the Charts tab. Thank you!

**Comment by ARGAMX** [comment]
Also From my point of view seems that the ChrizmanTV company's levels are not pretty balanced in terms of the hospital's cost requirements.

What I see in this save:

![Screenshot 2024-10-17 at 11 14 03](https://github.com/user-attachments/assets/c7c59fb6-9c2b-47ca-8b4d-a09d65ff700c)

What I was have to do to pass hospital's cost requirements on this level:

![Screenshot 2024-09-30 at 19 32 41](https://github.com/user-attachments/assets/84405081-6ebb-4a9c-a76b-ab35f902da70)

It's weird to me and makes the hospital look ugly. 😁

**Comment by Matroftt** [comment]
PR should be created which would make value requirements lower and fix death criteria
Also there is a bug in one of .map file from chrizTV campaign, which also need to fix (some of floor tiles are set as outside tiles)

**Comment by ARGAMX** [comment]
@xyressa why this topic closed? it real unresolved issue in the game.

**Comment by lewri** [comment]
To be fixed by #2676

**Comment by xyressa** [comment]
> @xyressa why this topic closed? it real unresolved issue in the game.

My bad, I thought it was just me being dumb and not realising I hadn't fulfilled all the win conditions, so I closed it.

Thanks for helping me spot the error and also being so quick to trigger a fix, even if it's just a minor error improvement!

**Comment by lewri** [comment]
Fixed by #2676

**Comment by ARGAMX** [comment]
I believe, Instead of showing fulfilled criterias, we should firstly show unfulfilled and first.
Priorities:
1. Critical criterias.
2. Unfulfilled criterias.
3. Fulfilled criterias.

By #2676 PR we have reduced the chance for this problem happening in this certain company. But in fact we have done nothing to prevent it from happening as such.

Can we please reopen this issue?
